### PR TITLE
replace NewTransaction() with NewTx()

### LINKED ISSUE #3990: replace NewTransaction() with NewTx()
### Description of the code refactor
`NewTransaction` and `NewContractCreation` are marked as deprecated in Ethereum, replace them with `NewTx`

### Why is this needed
- [] better/less memory usage
- [] reduce running time
- [] clean (easy-to-understand) data structure or algorithm
- [x] remove dead code
- [] better code design or structure (like reduce complexity)
- [] reduce/decouple dependencies between code modules
- [] increase test coverage
- [] better code readability

### PR DESCRIPTION
**Pull request by millken**
# Description
NewTransaction and NewContractCreation are marked as deprecated in Ethereum, replace them with NewTx

Fixes #3990

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Liuhaai — APPROVED** [review]

File: api/web3server.go [hunk 1]
```diff
@@ -401,13 +401,26 @@ func (svr *web3Handler) estimateGas(in *gjson.Result) (interface{}, error) {
...
 		}
-		tx = types.NewTransaction(0, toAddr, value, gasLimit, big.NewInt(0), data)
+		tx = types.NewTx(&types.LegacyTx{
+			Nonce:    0,
+			GasPrice: big.NewInt(0),
+			Gas:      gasLimit,
+			To:       &toAddr,
+			Value:    value,
+			Data:     data,
+		})
```
Comment (CoderZhi) [inline api/web3server.go:423]:
```
    var toAddr *common.Address
    if len(to) != 0 {
        toAddr, err := ...
        if err != nil {
            ....
        }
    }
    tx := types.NewTx(&types.LegacyTx{
	Nonce:    0,
	GasPrice: big.NewInt(0),
	Gas:      gasLimit,
	To:       toAddr,
	Value:    value,
	Data:     data,
    })
```

File: action/stake_reclaim.go [hunk 2]
```diff
@@ -305,5 +313,13 @@ func (sw *WithdrawStake) ToEthTx(_ uint32) (*types.Transaction, error) {
...
-	return types.NewTransaction(sw.Nonce(), _stakingProtocolEthAddr, big.NewInt(0), sw.GasLimit(), sw.GasPrice(), data), nil
+	tx := types.NewTx(&types.LegacyTx{
+		Nonce:    sw.Nonce(),
+		GasPrice: sw.GasPrice(),
+		Gas:      sw.GasLimit(),
+		To:       &_stakingProtocolEthAddr,
+		Value:    big.NewInt(0),
+		Data:     data,
+	})
+	return tx, nil
```
Comment (CoderZhi) [inline action/stake_reclaim.go:324]: ```return types.NewTx(...), nil```

File: action/stake_reclaim.go [hunk 3]
```diff
@@ -217,7 +217,15 @@ func (su *Unstake) ToEthTx(_ uint32) (*types.Transaction, error) {
 	if err != nil {
 		return nil, err
 	}
-	return types.NewTransaction(su.Nonce(), _stakingProtocolEthAddr, big.NewInt(0), su.GasLimit(), su.GasPrice(), data), nil
+	tx := types.NewTx(&types.LegacyTx{
```
Comment (CoderZhi) [inline action/stake_reclaim.go:220]: `return types.NewTx(...), nil`

File: action/stake_create.go [hunk 4]
```diff
@@ -266,5 +266,13 @@ func (cs *CreateStake) ToEthTx(_ uint32) (*types.Transaction, error) {
 	if err != nil {
 		return nil, err
 	}
-	return types.NewTransaction(cs.Nonce(), _stakingProtocolEthAddr, big.NewInt(0), cs.GasLimit(), cs.GasPrice(), data), nil
+	tx := types.NewTx(&types.LegacyTx{
```
Comment (CoderZhi) [inline action/stake_create.go:269]: `return types.NewTx(...), nil`

File: action/builder_test.go [hunk 5]
```diff
@@ -63,13 +78,27 @@ func TestBuildRewardingAction(t *testing.T) {
...
+		To:       &to,
+		Value:    big.NewInt(100),
+		Data:     depositData,
+	})
 	env, err = eb.BuildRewardingAction(tx)
 	r.Nil(env)
 	r.EqualValues("invalid action type", err.Error())
 
-	tx = types.NewTransaction(1, _rewardingProtocolEthAddr, big.NewInt(100), 10000, big.NewInt(10004), depositData)
+	tx = types.NewTx(&types.LegacyTx{
```
Comment (CoderZhi) [inline action/builder_test.go:94]: `env, err = eb.BuildRewardingAction(types.NewTx(...))`

File: action/builder_test.go [hunk 6]
```diff
@@ -63,13 +78,27 @@ func TestBuildRewardingAction(t *testing.T) {
...
+	to = common.HexToAddress("0x0000000000000000000000000000000000000001")
+	tx = types.NewTx(&types.LegacyTx{
+		Nonce:    1,
+		GasPrice: big.NewInt(10004),
+		Gas:      10000,
+		To:       &to,
+		Value:    big.NewInt(100),
+		Data:     depositData,
+	})
 	env, err = eb.BuildRewardingAction(tx)
```
Comment (CoderZhi) [inline action/builder_test.go:90]: same here

**Review by CoderZhi — APPROVED** [review]

File: api/web3server.go [hunk 7]
```diff
@@ -400,15 +401,22 @@ func (svr *web3Handler) estimateGas(in *gjson.Result) (interface{}, error) {
 	}
 
 	var tx *types.Transaction
-	if len(to) == 0 {
-		tx = types.NewContractCreation(0, value, gasLimit, gasPrice, data)
-	} else {
-		toAddr, err := addrutil.IoAddrToEvmAddr(to)
+	var toAddr *common.Address
```
Comment (dustinxie) [inline api/web3server.go:404]: var ()

File: action/execution.go [hunk 8]
```diff
@@ -235,36 +235,32 @@ func (ex *Execution) SanityCheck() error {
...
-				AccessList: ex.accessList,
-			}), nil
+	var ethAddr *common.Address
+	if ex.contract != EmptyAddress {
+		addr, err := address.FromString(ex.contract)
+		if err != nil {
+			return nil, err
 		}
-		return types.NewContractCreation(ex.Nonce(), ex.amount, ex.GasLimit(), ex.GasPrice(), ex.data), nil
+		*ethAddr = common.BytesToAddress(addr.Bytes())
```
Comment (dustinxie) [inline action/execution.go:244]: `ethAddr = &common.BytesToAddress(addr.Bytes())` is better?

Comment (millken) [reply action/execution.go:244]: can not direct do this , you must define a middle variable
```
		aa := common.BytesToAddress(addr.Bytes())
		ethAddr = &aa
```

**Review by dustinxie — APPROVED** [review]

**Pull request merged**